File: numbers/src/main/java/org/dicio/numbers/parser/param/ExtractDurationParams.java
package org.dicio.numbers.parser.param;

import org.dicio.numbers.parser.Parser;
import org.dicio.numbers.parser.lexer.TokenStream;
import org.dicio.numbers.unit.Duration;

import java.util.function.Supplier;

public class ExtractDurationParams extends NumberParserParams<Duration> {
    // TODO documentation
    // default values
    private boolean shortScale = true;

    public ExtractDurationParams(final Parser parser, final String utterance) {
        super(parser, utterance);
    }

    /**
     * @param shortScale consider large numbers that make up durations (e.g. the "one billion" in
     *                   "one billion milliseconds") as short scale (true) or long scale (false)
     *                   numbers (see <a
     *                   href="https://en.wikipedia.org/wiki/Names_of_large_numbers">Names of large
     *                   numbers</a>). The default is true.
     * @return this
     */
    public ExtractDurationParams shortScale(final boolean shortScale) {
        this.shortScale = shortScale;
        return this;
    }

    @Override
    protected Supplier<Duration> getExtractorAtCurrentPosition(final TokenStream tokenStream) {
        return parser.extractDuration(tokenStream, shortScale);
    }
}

File: numbers/src/main/java/org/dicio/numbers/parser/param/ExtractDurationParams.kt
package org.dicio.numbers.parser.param

import org.dicio.numbers.parser.Parser
import org.dicio.numbers.parser.lexer.TokenStream
import org.dicio.numbers.unit.Duration

class ExtractDurationParams(parser: Parser, utterance: String) :
    ParserParams<Duration>(parser, utterance) {
    // default values
    private var shortScale = true

    /**
     * @param shortScale consider large numbers that make up durations (e.g. the "one billion" in
     * "one billion milliseconds") as short scale (true) or long scale (false) numbers (see
     * [Names of large numbers](https://en.wikipedia.org/wiki/Names_of_large_numbers)). **The
     * default is `true`.** This setting only affects languages where both short and long scale
     * numbers exist, e.g. English, and is otherwise ignored.
     * @return this
     */
    fun shortScale(shortScale: Boolean): ExtractDurationParams {
        this.shortScale = shortScale
        return this
    }

    override fun getExtractorAtCurrentPosition(tokenStream: TokenStream): () -> Duration? {
        return parser.extractDuration(tokenStream, shortScale)
    }
}
